isodatetime/data.py

```diff
@@ -1315,25 +1332,137 @@ def get_props(self):
+        other = other.copy()
+        other.set_time_zone(self.get_time_zone())
+        if self.get_is_calendar_date():
+            my_date = self.get_calendar_date()
+            other_date = other.get_calendar_date()
+        else:
+            my_date = self.get_ordinal_date()
+            other_date = other.get_ordinal_date()
+        my_datetime = list(my_date) + [self.get_second_of_day()]
+        other_datetime = list(other_date) + [other.get_second_of_day()]
+        return my_datetime < other_datetime
+
+    def __le__(self, other: "TimePoint") -> bool:
+        if other is None:
+            return False
+        if self.truncated != other.truncated:
+            raise TypeError(
+                "Cannot compare truncated to non-truncated " +
+                "TimePoint: %s, %s" % (self, other))
+        if self.get_props() == other.get_props():
+            return True
+        if self.truncated:
+            for attribute in self.DATA_ATTRIBUTES:
+                other_attr = getattr(other, attribute)
+                self_attr = getattr(self, attribute)
+                if self_attr != other_attr:
+                    return self_attr <= other_attr
+            return True
+        other = other.copy()
+        other.set_time_zone(self.get_time_zone())
+        if self.get_is_calendar_date():
+            my_date = self.get_calendar_date()
+            other_date = other.get_calendar_date()
+        else:
+            my_date = self.get_ordinal_date()
+            other_date = other.get_ordinal_date()
+        my_datetime = list(my_date) + [self.get_second_of_day()]
+        other_datetime = list(other_date) + [other.get_second_of_day()]
+        return my_datetime <= other_datetime
+
+    def __gt__(self, other: "TimePoint") -> bool:
+        if other is None:
+            return True
+        if self.truncated != other.truncated:
+            raise TypeError(
+                "Cannot compare truncated to non-truncated " +
+                "TimePoint: %s, %s" % (self, other))
+        if self.get_props() == other.get_props():
+            return False
+        if self.truncated:
+            for attribute in self.DATA_ATTRIBUTES:
+                other_attr = getattr(other, attribute)
+                self_attr = getattr(self, attribute)
+                if self_attr != other_attr:
+                    return self_attr > other_attr
+            return True
+        other = other.copy()
+        other.set_time_zone(self.get_time_zone())
+        if self.get_is_calendar_date():
+            my_date = self.get_calendar_date()
+            other_date = other.get_calendar_date()
+        else:
+            my_date = self.get_ordinal_date()
+            other_date = other.get_ordinal_date()
+        my_datetime = list(my_date) + [self.get_second_of_day()]
+        other_datetime = list(other_date) + [other.get_second_of_day()]
+        return my_datetime > other_datetime
+
+    def __ge__(self, other: "TimePoint") -> bool:
```

Comment:
I wonder if we can just do `return not self.__lt__(other)`?

Reply:
Maybe for the `__gt__`. `functools.total_ordering` would simplify it, but at the cost of performance... so maybe we can implement

- `__eq__`
- `__lt__`

And return `not __eq__` for `__ne__`, `__eq__ and __lt__` for `__le__`, and so it goes? I read somewhere that `__ne__` would call `not __eq__` by default, and wasn't mandatory to implement it, but haven't tried it.

Reply:
We could just implement `__eq__` and `__lt__` as you suggest, there is no need to do `not __eq__` type stuff. If it's quick to do we should probably just go for it and implement the full array for the performance improvement though?

Reply:
Resolving the conversation as we have dealt with this problem already :tada: